readconf.c

```diff
@@ -1727,25 +1731,57 @@ read_config_file_depth(const char *filename, struct passwd *pw,
 	FILE *f;
 	char line[4096];
 	int linenum;
-	int bad_options = 0;
+	int bad_options = 0;	
 
 	if (depth < 0 || depth > READCONF_MAX_DEPTH)
 		fatal("Too many recursive configuration includes");
 
-	if ((f = fopen(filename, "r")) == NULL)
+	if ((f = fopen(filename, "r")) == NULL) 
 		return 0;
-
-#ifndef WINDOWS /* TODO - implement permission checks for Windows */
 	if (flags & SSHCONF_CHECKPERM) {
+		
+#if WINDOWS
+		HANDLE h;
+		PSID owner_sid = NULL;		
+		DWORD ret;
+		PSECURITY_DESCRIPTOR pSD = NULL;
+		struct _stat sb;
+
+		if (_stat(filename, &sb) == -1)
+			fatal("_stat %s: %s", filename, strerror(errno));
+
+		h = (HANDLE)_get_osfhandle(_fileno(f));
+		if (h == -1 && errno == EBADF)
+			fatal("failed to retrieve the handle of file %s", filename);
+
+		/*Get the owner SID of the file.*/
+		ret = GetSecurityInfo(
+			h,
+			SE_FILE_OBJECT,
+			OWNER_SECURITY_INFORMATION,
+			&owner_sid,
+			NULL,
+			NULL,
+			NULL,
+			&pSD);
+		if (ret != ERROR_SUCCESS)			
+			fatal("failed to retrieve the owner sid of the file %s with errorcode %d", filename, ret);		
+					
+		if (EqualSid(owner_sid, getusid()) == FALSE ||
```

Comment:
I don't understand. if "info" is freed does that free up User.Sid too? If so, we are using a deleted reference. 
If not, we need to explicity delete User.Sid.